blackjack.py

```diff
@@ -242,11 +269,30 @@ def play_again():
             continue
 
 
-def retotal(h):
-    new_total = 0
-    for card in h:
-        new_total += card_point(card)
-    return new_total
+def hand_total(hand):
+    total = 0
+
+    # Evaluate card points and add them to total.
+    for card in hand:
+        r = card.rank
+        points = card_point(card)
+
+        # If an Ace (11) would cause a bust, change to 1. If not Ace remains 11.
+        if r == Rank.Ace:
+            if total + points > 21:
+                total -= 10
+        total += points
```

Comment:
This is a little confusing because you do `total -= 10` before you've done `total += points`. It would be a little clearer to do it above the `if r == Rank.Ace` check. In fact, you could just do something like `total += card_point(card)` and not create a `points` variable at all.

Also, you can combine the two `if` statements with `and`; e.g. `if r == Rank.Ace and total >= 21`.

Reply:
When I change the order like you suggest [A, A] returns 2.

Reply:
resolved